sql/index_test.go

```diff
@@ -0,0 +1,324 @@
+package sql
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/shopspring/decimal"
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestRoundUpFloat(t *testing.T) {
```

Comment:
Will this this maybe need to generalize to other types at some point? We don't need to add anything for strings, blobs, enums, etc, but it would be nice if additional bugs in this area folded into this impl/testing strategy